### ISSUE TITLE
Implement Open Graph protocol to increase social media presence

### ISSUE THREAD

**Issue by majordwarf** [MovingBlocks/movingblocks.github.com#25]
Open Graph protocol (OGP) gives you control of how website/page specifc data travels to third-party sites.  Providing much better social media presence along with optimized SEO. Right now we are using Helmet to do manage the head and meta tags for the pages. The main `layout` component would require some restructuring along with additional changes to other pages to implement OGP as currently the `head` is being rendered inside body, which _works_ but is a bad practice. 

You can learn more about OGP [here](https://ogp.me/).

---
***Things to be done:***
- [ ] Restructure `layout` component and other pages to place head section appropriately.
- [ ] Implement OGP on the existing pages.
- [ ] Extend OGP to blog posts and modules.

**Comment by ANURADHAJHA99** [comment]
I would like to work on the following issue. Thank you very much.

**Comment by GraceChiamaka** [comment]
@ANURADHAJHA99 are you working on this?

**Comment by ANURADHAJHA99** [comment]
@GraceChiamaka Yes, I am working on it.

**Comment by ANURADHAJHA99** [comment]
@majordwarf I was going through the layout component and started working on introducing OGP, but I am having certain issues understanding how should I introduce the basic metadata? Can you please provide me a small example, or help me understand it?

**Comment by majordwarf** [comment]
You can look at how meta-data for blog and modules are stored in their respective markdown files as frontmatter. We need to implement OGP to take that data dynamically to make the shared post on social media with appropriate metadata being shown.

**Comment by ANURADHAJHA99** [comment]
can we use OGP in markdown modules?

**Comment by ANURADHAJHA99** [comment]
`import React from "react";
import "./About.css";

export default () => (
  <div className="about">
    <h1>About the game.</h1>
    <meta property="og:title" content="about" />
  </div>
);
`
This is the "about" component that I am working on, I have inserted an OGP metadata, am i doing it right?

**Comment by majordwarf** [comment]
Nope, we currently have 200+ modules. Updating each `README.md` with their respective meta-data manually is a tedious task. The main purpose of this project was to automate as many as tasks possible and to reduce manual effort. Also, the OpenGraph tags won't be generated inside the markdown file. it's just the wrong heirarchy of tags even if they are generated. We want the tags to be generated inside the head of every page. The content of the markdown is used to go inside the body not inside the head.

**Comment by ANURADHAJHA99** [comment]
![Screenshot from 2019-10-14 15-51-50](https://user-images.githubusercontent.com/34815869/66744642-97c2a180-ee9a-11e9-9adf-1a84e1eef581.png)




I updated the about component, with basic metadata, am I doing it correctly?

**Comment by majordwarf** [comment]
No, the only thing that needs to change is the `layout` component and all the other pages, posts pages passing props back to parent to avoid writing the same boilerplate code again and again.

So the `layout` will contain variables that will replace the OpenGraph meta-tags' content. During the build, the child component will pass some variables back to the parent to replace the variables.

**Comment by ANURADHAJHA99** [comment]
Ok, got it now! Thank you very much for your help.

**Comment by ANURADHAJHA99** [comment]
Sorry, I have a doubt, we are suppose to pass the  child header component with the variables in the layout component such that it replaces the variable in the header parent component?

**Comment by ANURADHAJHA99** [comment]
![Screenshot from 2019-10-14 19-14-39](https://user-images.githubusercontent.com/34815869/66756242-378e2880-eeb7-11e9-86e4-1929741427b2.jpg)

we have to add states within the highlighted component? I am sorry to continuously disturb, I am learning and working on it.

**Comment by majordwarf** [comment]
This layout contains Helmet as stated in the first message of the issue itself. It handles how meta-data tags are handled on the website. Currently all I had in mind was to pass props from child like the blog template, post template to the current layout component and that will fill out the variables in Helment component. But not having actually work on it, it might still needs some research. If passing props won't work we might have to rewrite the templates written already.

**Comment by ANURADHAJHA99** [comment]
so we are suppose to change the props in the blog.jsx and post.jsx?

**Comment by ANURADHAJHA99** [comment]
Essentially I meant, we are supposed to introduce OGP metadata in the blog template and post template?

**Comment by majordwarf** [comment]
Yes but if you want to directly implement OGP in their respective template, they will need restructuring. Eliminating the need for layout component in them. Or else pass some props to layout if you're using the layout component.

**Comment by ANURADHAJHA99** [comment]
There are many other components within the layout components, what should I do to them?

**Comment by majordwarf** [comment]
Every child component needs to have the tags as said earlier.
